src/relations/postgresql_provider.py

```diff
@@ -153,27 +152,16 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
                 )
             )
 
-    def _on_relation_changed(self, event: RelationChangedEvent) -> None:
```

Comment:
Checking for pg_hba changes before releasing `database_requested` should make this hook redundant.